cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -124,7 +120,7 @@ public final ClusterStatus deleteClusterWithStatus(ProvisionerContext context, C
         jobManager.close();
       }
 
-      String bucket = DataprocUtils.getBucketName(properties.get(BUCKET));
+      String bucket = conf.getGcsBucket() != null ? conf.getGcsBucket() : properties.get(DataprocUtils.BUCKET);
```

Comment:
nit - Add a comment explaining due to historical reasons `bucket` is used if `conf.getGcsBucket` is not provided.

Reply:
Done